File: java/03_library_native_jni/source/src/test/java/namespace/StringComparatorTest.java
${{VAR_COPYRIGHT_HEADER}}

${{VAR_NAMESPACE_PACKAGE_DECLARATION}}

import static org.junit.Assert.*;
import static org.junit.Assume.*;

import org.junit.After;
import org.junit.AfterClass;
import org.junit.Before;
import org.junit.BeforeClass;
import org.junit.Test;

/**
 * Tests for the StringComparator class.
 *
 */
public class StringComparatorTest {

    /**
    * Indicates whether native shared libraries have been built
    * and are available to the test suite.
    */
    public static boolean nativeAvailable;

    @BeforeClass
    public static void setUpBeforeClass(){
        nativeAvailable = System.getProperty("nativeLibsAvailable", "false")
                                .equals("true");
    }

    @AfterClass
    public static void tearDownAfterClass(){ }

    @Before
    public void setUp(){ }

    @After
    public void tearDown(){ }

    @Test
    public void testTrivial() throws Exception{
        assertTrue(1 == 1);
    }

    @Test
    public void testStringEqual() throws Exception{
        assumeTrue(nativeAvailable);
        Comparator<String> dummy = new StringComparator("TEST-C");
        int res = dummy.compare("TEST-C");
        assertTrue(res == 0);
    }

    @Test
    public void testStringGreater() throws Exception{
        assumeTrue(nativeAvailable);
        Comparator<String> dummy = new StringComparator("TEST-C");
        int res = dummy.compare("TEST-E");
        assertTrue(res == -2);
    }

    @Test
    public void testStringSmaller() throws Exception{
        assumeTrue(nativeAvailable);
        Comparator<String> dummy = new StringComparator("TEST-C");
        int res = dummy.compare("TEST-A");
        assertTrue(res == 2);
    }

}

File: java/03_library_native_jni/source/src/test/cpp/namespace/StringComparatorTest.cpp
${{VAR_COPYRIGHT_HEADER}}

#include "gtest/gtest.h"
#include "gmock/gmock.h"

#include "${{VAR_NAMESPACE_PATH}}/StringComparator.h"


using std::string;
using ${{VAR_NAMESPACE_COLON}}::StringComparator;


int main(int argc, char** argv){
    ::testing::InitGoogleTest(&argc, argv);
    ::testing::InitGoogleMock(&argc, argv);
    return RUN_ALL_TESTS();
}

TEST(StringComparatorTest, TestTrivial){
    ASSERT_EQ(1, 1);
}

TEST(StringComparatorTest, TestCompareEqual){
    StringComparator dummy;
    int result = dummy.compare("TEST-1", "TEST-1");
    ASSERT_EQ(result, 0);
}
